tests/pipelines/test_transform_step.py

```diff
@@ -35,33 +36,55 @@ def set_up_transform_step(pipeline_root: Path):
     dataset.to_parquet(str(split_step_output_dir / "train.parquet"))
 
     pipeline_yaml = pipeline_root.joinpath(_PIPELINE_CONFIG_FILE_NAME)
+    experiment_name = "demo"
+    MlflowClient().create_experiment(experiment_name)
+
     pipeline_yaml.write_text(
         """
         template: "regression/v1"
         target_col: "y"
         experiment:
-          name: "demo"
+          name: {experiment_name}
           tracking_uri: {tracking_uri}
         steps:
           transform:
             transformer_method: sklearn.preprocessing.StandardScaler
         """.format(
-            tracking_uri=mlflow.get_tracking_uri()
+            tracking_uri=mlflow.get_tracking_uri(), experiment_name=experiment_name
         )
     )
     pipeline_config = read_yaml(pipeline_root, _PIPELINE_CONFIG_FILE_NAME)
     transform_step = TransformStep.from_pipeline_config(pipeline_config, str(pipeline_root))
-    return transform_step, transform_step_output_dir
+    return transform_step, transform_step_output_dir, split_step_output_dir
 
 
 def test_transform_step_writes_onehot_encoded_dataframe_and_transformer_pkl(tmp_pipeline_root_path):
     with mock.patch.dict(
         os.environ, {_MLFLOW_PIPELINES_EXECUTION_DIRECTORY_ENV_VAR: str(tmp_pipeline_root_path)}
     ):
-        transform_step, transform_step_output_dir = set_up_transform_step(tmp_pipeline_root_path)
+        transform_step, transform_step_output_dir, _ = set_up_transform_step(tmp_pipeline_root_path)
         transform_step._run(str(transform_step_output_dir))
 
     assert os.path.exists(transform_step_output_dir / "transformed_training_data.parquet")
     transformed = pd.read_parquet(transform_step_output_dir / "transformed_training_data.parquet")
     assert len(transformed.columns) == 3
     assert os.path.exists(transform_step_output_dir / "transformer.pkl")
+
+
+def test_transform_empty_step(tmp_pipeline_root_path):
+    with mock.patch.dict(
+        os.environ, {_MLFLOW_PIPELINES_EXECUTION_DIRECTORY_ENV_VAR: str(tmp_pipeline_root_path)}
+    ), mock.patch("sklearn.preprocessing.StandardScaler", return_value=None):
```

Comment:
Can we mock the user `transformer_fn()` to mimic the expected user input instead of mocking the `StandardScaler`?